app/controllers/application_controller.rb

```diff
@@ -7,4 +8,122 @@ class ApplicationController < Sinatra::Base
     set :views, 'app/views'
   end
 
+  enable :sessions
+
+  get '/tweets/new' do 
+    redirect :'/login' if !session[:id]
+    erb "<form action='/tweets/new' method='post'>
+    <input type='text' name='content'>Tweet</input>
+    <input type='submit' id='submit'>
+    </form>"
+  end
+
+  post '/tweets/new' do 
+    redirect :'/tweets/new' if params[:content].blank?
+    user = User.find(session[:id])
+    user.tweets.build(params)
+    user.save
+  end
+
+  get '/' do
+    erb 'Welcome to Fwitter'
+  end
+
+  get '/signup' do 
+    redirect :'/tweets' if session[:id]
+  end
+
+  post '/signup' do
+    redirect :'/signup' if params[:username]==""
+    redirect :'/signup' if params[:email]=="" 
+    redirect :'/signup' if params[:password]==""
+    
+    user = User.create(params)
+    session[:id] = user.id
+
+    redirect :'/tweets'
+  end
+
+  get '/login' do
+
+    redirect :'/tweets' if session[:id]
+    
+    erb "<%= session[:id].nil? %>
```

Comment:
It's normally best practice to render an erb file rather than have your markup in the controller.